### ISSUE TITLE
Look into report of prop ATTR_X_SCL, ATTR_Y_SCL bug

### ISSUE THREAD

**Issue by patrickmeehan** [moai/moai-dev#102]
From Bonozo:

Just fyi, I'm showing that for a Prop2D...

attachedProp:setAttrLink( MOAIProp2D.INHERIT_TRANSFORM, prop, MOAIProp2D.TRANSFORM_TRAIT )

works great, but...

attachedProp:setAttrLink:setAttrLink( MOAITransform.ATTR_X_SCL,  prop , MOAITransform.ATTR_WORLD_X_SCL )
attachedProp:setAttrLink:setAttrLink( MOAITransform.ATTR_Y_SCL,  prop , MOAITransform.ATTR_WORLD_Y_SCL )

and the scale is exaggerated, i.e. the attachedProp is larger or smaller than "prop".  

Both images are same resolution ("prop" and "attachedProp")

**Comment by patrickmeehan** [comment]
More from Bonozo:

ok just fyi, if  TRANSFORM_TRAIT and  ATTR_WORLD_X_SCL / ATTR_WORLD_Y_SCL linked together, causes the bug.

If I link just TRANFORM and use setScl, fine.  Link SCL and thus use setLoc, fine.  setLoc/setScl, of course, fine.  Link TRANSFORM and SCL, unexpected results.

**Comment by danadrew** [comment]
I added a sample: samples / test / prop-attr-link

**Expected:**
I should be able to set the frame of two props and link their scales.

**Actual:**
If I set the frame _and_ link a scale attribute, then the child prop's scale will be much larger than expected.

**Additional Info:**
If I only set the frame or link the scale then it behaves as expected, it's only when I do both that it breaks. When it breaks, the scale of the child is being set to half the width of the parent. Not that the size is getting set to that, but the scale, so it ends up being scaled quite large.
